### PR TITLE
Fix invalid signing

### PR DESCRIPTION
**Pull request by xPi2**
**Post-mortem**
- #111 Added a hook to show "Approve transfers" button based on certain conditions.
- #101 Used a placeholder Permit with nonce 0 as quick solution in the hook until the fetch of the next nonce available for Permit2 signature allowance, which caused the configuration for signing to load twice.
- Missing dependency of `signPermit()` function in the "Approve transfers" button hook caused it to not reload properly, leading to a scenario where the user was prompted with a different Permit to sign than the one that would be sent to the contract, breaking the join.

**Solution**
- Omitted placeholder Permit and initially load `signPermit()` with incomplete configuration to prevent prompting the user to sign a placeholder Permit.
- Added a `signReady` flag to signal when the Permit is ready for signature and included it in the button hook to prevent initiating signing before the Permit is ready.

**TL;DR:**
We fixed the issue where users were prompted to sign a placeholder Permit by omitting it and adding a `signReady` flag to ensure signing only happens when the Permit is ready.

### CODE REVIEW

**Review by gordo-labs — APPROVED** [review]

**Review by q1keonchain — COMMENTED** [review]
Interesting edge case; also good find 👍
